src/transaction/TransactionController.ts

```diff
@@ -669,12 +674,12 @@ export class TransactionController extends BaseController<TransactionConfig, Tra
   async queryTransactionStatuses() {
     const { transactions } = this.state;
     const network = this.context.NetworkController;
-    const currentNetworkID = network.state.provider.chainId;
+    const currentChainId = network.state.provider.chainId;
     let gotUpdates = false;
     await safelyExecute(() =>
       Promise.all(
         transactions.map(async (meta, index) => {
-          if (meta.status === 'submitted' && meta.networkID === currentNetworkID) {
+          if (meta.status === 'submitted' && (meta.networkID === currentChainId || meta.chainId === currentChainId)) {
```

Comment:
Interesting :thinking: So this was already using the `chainId`, resulting in the wrong set of transactions being returned whenever the `chainId` was different from the `networkId`.